index.js

```diff
@@ -0,0 +1,213 @@
+'use strict';
+
+var thrift = require('thrift');
+var through = require('through2');
+var Types = require('./src/bolt_types');
+var BoltManagerService = require('./src/BoltManagerService');
+var BoltProxyService = require('./src/BoltProxyService');
+var BoltSchedulerService = require('./src/BoltSchedulerService');
+
+var ComputationMetadata = Types.ComputationMetadata;
+var StreamMetadata = Types.StreamMetadata;
+//
+// Transation returned on each request
+//
+var Tx = Types.ComputationTx;
+var Record = Types.Record;
+var Endpoint = Types.Endpoint;
+
+
+function Computation(options) {
+  //
+  // Remark: Do we want to fail if we dont get a name? For now lets generate
+  // a unique identifier
+  //
+  this.name = options.name || 'node-' + process.hrtime().join('-');
+  //
+  // Set the grouping to the correct INT code based on what we have in the
+  // thrift types. We default to SHUFFLE
+  //
+  this.grouping = groupMap(options.grouping || 'shuffle');
```

Comment:
I love the idea of having the 'metadata' be part of the constructor object. :+1: 

However, I would remove the ` this.grouping ` as it doesn't make sense to have a grouping for ALL streams